### DISCUSSION TITLE
'assignRef' is not exported from '@mantine/hooks' when used in component library

### DISCUSSION THREAD

**Discussion by ShortRoundDev** [mantinedev/mantine#6691] [Q&A] [answered]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.12.1

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

Not applicable – issue is not related to the browser

### Describe the bug

I have a component library (Vite) which uses mantine under the hood. my component library builds just fine, but when I try to import it into my client application (react-app-rewired) I get:

```
./node_modules/@mantine/core/esm/components/Portal/Portal.mjs
Attempted import error: 'assignRef' is not exported from '@mantine/hooks'.
```

I have attempted to build my component library with mantine as a dependency, and I've tried it as a peer dependency, but I still get this issue. My client library is built as cjs and both @mantine/core and @mantine/hooks are the same version (7.12.1)

### Self-service

- [x] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment, marked as answer]
It is outside of Mantine scope, the `@mantine/hooks` package itself provides `assignRef` import. The issue is probably with your app or bundled package. If you want to build a custom package and publish it to npm, see example template – https://github.com/mantinedev/extension-template

**Reply by ShortRoundDev** [reply]
Thanks for replying. It turns out that my client application had cached a copy of mantine from when I had it set as a peer dependency. Removing mantine manually from `node_modules` fixed that issue, and then I had a set of other problems I needed to solve. Here are the things I needed to do in my repo, for anyone in the future:

1. Mantine must not be imported by the client; it must be referenced by the component library (i.e: it is a dependency and not a peerDependency)

2. I am running my component library with yarn link locally, so react must be aliased in the react-app-rewired config overrides so that the component library's react import is the same as the frontend react import (cannot mix react runtimes. This solves "dispatch is null")

3. The component library must wrap a MantineProvider and export it so that I can instantiate that mantine provider without adding mantine as a direct dependency to the client (see no. 1)

4. I upgraded react in my client to ^18.x.x. this solves other react issues like "dispatch.useId is undefined"

So if any else in the future is doing what I'm doing and is running into any of the same issues, hopefully this helps
